pkg/controller/container-runtime-config/container_runtime_config_controller_test.go

```diff
@@ -1607,3 +1691,59 @@ func TestCleanUpDuplicatedMC(t *testing.T) {
 		})
 	}
 }
+
+func TestClusterImagePolicyCreate(t *testing.T) {
+	for _, platform := range []apicfgv1.PlatformType{apicfgv1.AWSPlatformType, apicfgv1.NonePlatformType, "unrecognized"} {
+		t.Run(string(platform), func(t *testing.T) {
+			f := newFixture(t)
+
+			cc := newControllerConfig(ctrlcommon.ControllerConfigName, platform)
+			mcp := helpers.NewMachineConfigPool("master", nil, helpers.MasterSelector, "v0")
+			mcp2 := helpers.NewMachineConfigPool("worker", nil, helpers.WorkerSelector, "v0")
+			imgcfg1 := newImageConfig("cluster", &apicfgv1.RegistrySources{InsecureRegistries: []string{"blah.io"}, AllowedRegistries: []string{"allow.io"}, ContainerRuntimeSearchRegistries: []string{"search-reg.io"}})
+
+			cvcfg1 := newClusterVersionConfig("version", "test.io/myuser/myimage:test")
+			keyReg1, _ := getManagedKeyReg(mcp, nil)
+			keyReg2, _ := getManagedKeyReg(mcp2, nil)
+			clusterimgPolicyKey1 := getManagedKeyNamespacedImagePolicy(mcp)
+			clusterimgPolicyKey2 := getManagedKeyNamespacedImagePolicy(mcp)
+			clusterimgPolicyMCs1 := helpers.NewMachineConfig(clusterimgPolicyKey1, map[string]string{"node-role": "master"}, "dummy://", []ign3types.File{{}})
+			clusterimgPolicyMCs2 := helpers.NewMachineConfig(clusterimgPolicyKey2, map[string]string{"node-role": "worker"}, "dummy://", []ign3types.File{{}})
+
+			mcs1 := helpers.NewMachineConfig(keyReg1, map[string]string{"node-role": "master"}, "dummy://", []ign3types.File{{}})
+			mcs2 := helpers.NewMachineConfig(keyReg2, map[string]string{"node-role": "worker"}, "dummy://", []ign3types.File{{}})
+			clusterimgPolicy := newClusterImagePolicyWithPublicKey("image-policy", []string{"example.com"}, "Zm9vIGJhcg==")
+			f.ccLister = append(f.ccLister, cc)
+			f.mcpLister = append(f.mcpLister, mcp)
+			f.mcpLister = append(f.mcpLister, mcp2)
+			f.imgLister = append(f.imgLister, imgcfg1)
+			f.clusterImagePolicyLister = append(f.clusterImagePolicyLister, clusterimgPolicy)
+			f.cvLister = append(f.cvLister, cvcfg1)
+			f.imgObjects = append(f.imgObjects, imgcfg1)
+
+			f.expectGetMachineConfigAction(mcs1)
+			f.expectGetMachineConfigAction(mcs1)
+			f.expectGetMachineConfigAction(mcs1)
+			f.expectCreateMachineConfigAction(mcs1)
+			f.expectGetMachineConfigAction(clusterimgPolicyMCs1)
+			f.expectCreateMachineConfigAction(clusterimgPolicyMCs1)
+
+			f.expectGetMachineConfigAction(mcs2)
+			f.expectGetMachineConfigAction(mcs2)
+			f.expectGetMachineConfigAction(mcs2)
+
+			f.expectCreateMachineConfigAction(mcs2)
+			f.expectGetMachineConfigAction(clusterimgPolicyMCs2)
+			f.expectCreateMachineConfigAction(clusterimgPolicyMCs2)
```

Comment:
No expectation of a CIP access?

(I would also think that the existing tests should now start accessing CIP, when the feature gate is hard-coded to on. … and maybe there should be tests that the feature gate works.)

Reply:
current action verification tests only check the machineconfiguration client, it does not include the config, operator API client, and the list of resources intended to be filtered out. We can improve the test to include other client, but not in this PR.
https://github.com/openshift/machine-config-operator/blob/53a9e00f77a6bfa2a4364b7490c03f292f32425d/pkg/controller/container-runtime-config/container_runtime_config_controller_test.go#L309-L312

Reply:
I think this is the featuregate tests https://github.com/openshift/machine-config-operator/blob/master/pkg/controller/kubelet-config/kubelet_config_features_test.go